usvm-ts/src/main/kotlin/org/usvm/machine/TSContext.kt

```diff
@@ -0,0 +1,193 @@
+package org.usvm.machine
+
+import io.ksmt.sort.KFp64Sort
+import io.ksmt.utils.asExpr
+import org.jacodb.ets.base.EtsBooleanType
+import org.jacodb.ets.base.EtsNumberType
+import org.jacodb.ets.base.EtsRefType
+import org.jacodb.ets.base.EtsType
+import org.jacodb.ets.base.EtsUnknownType
+import org.jacodb.ets.model.EtsScene
+import org.usvm.UAddressSort
+import org.usvm.UBoolExpr
+import org.usvm.UBoolSort
+import org.usvm.UBv32Sort
+import org.usvm.UConcreteHeapRef
+import org.usvm.UContext
+import org.usvm.UExpr
+import org.usvm.UHeapRef
+import org.usvm.USort
+import org.usvm.api.makeSymbolicPrimitive
+import org.usvm.collection.field.UFieldLValue
+import org.usvm.isFalse
+import org.usvm.machine.expr.TSUndefinedSort
+import org.usvm.machine.expr.TSUndefinedValue
+import org.usvm.machine.expr.TSUnresolvedSort
+import org.usvm.machine.interpreter.TSStepScope
+import org.usvm.machine.types.ExprWithTypeConstraint
+import org.usvm.machine.types.FakeType
+import org.usvm.types.single
+import kotlin.contracts.ExperimentalContracts
+import kotlin.contracts.contract
+
+typealias TSSizeSort = UBv32Sort
+
+class TSContext(
+    val scene: EtsScene,
+    components: TSComponents,
+) : UContext<TSSizeSort>(components) {
+    val undefinedSort: TSUndefinedSort by lazy { TSUndefinedSort(this) }
+
+    val unresolvedSort: TSUnresolvedSort = TSUnresolvedSort(this)
+
+    private val undefinedValue by lazy { TSUndefinedValue(this) }
+
+    fun typeToSort(type: EtsType): USort = when (type) {
+        is EtsBooleanType -> boolSort
+        is EtsNumberType -> fp64Sort
+        is EtsRefType -> addressSort
+        is EtsUnknownType -> unresolvedSort
+        else -> TODO("Support all JacoDB types, encountered $type")
+    }
+
+    // TODO fix conjuncts
+    fun mkTruthyExpr(expr: UExpr<out USort>, scope: TSStepScope): UBoolExpr = scope.calcOnState {
```

Comment:
I don't like logic with `TSStepScope` inside the Context class. Maybe it is better to move it somewhere else, e.g. as extension function